tests/integration/components/bs-modal-test.js

```diff
@@ -246,4 +246,51 @@ module('Integration | Component | bs-modal', function (hooks) {
 
     assert.dom('.modal').isVisible('Modal is visible again');
   });
+
+  testBS3('it animates opening and closing the modal', async function (assert) {
+    this.set('open', false);
+    await render(hbs`<BsModal @open={{this.open}}>Hello world!</BsModal>`);
+
+    this.set('open', true);
+    await waitFor('.modal.show');
+    assert.dom('.modal').hasClass('show');
+
+    await settled();
+    assert.dom('.modal').hasClass('show');
+    assert.dom('.modal').hasClass('in');
+
+    this.set('open', false);
+    await waitUntil(() => {
+      return !find('.modal').classList.contains('in');
+    });
+    assert.dom('.modal').hasClass('show');
+    assert.dom('.modal').doesNotHaveClass('in');
+
+    await settled();
+    assert.dom('.modal').doesNotExist();
+  });
+
+  testBS4('it animates opening and closing the modal', async function (assert) {
+    this.set('open', false);
+    await render(hbs`<BsModal @open={{this.open}}>Hello world!</BsModal>`);
+
+    this.set('open', true);
+    await waitFor('.modal.show');
```

Comment:
I should have commented here in the first place, but did in the other PR: https://github.com/kaliber5/ember-bootstrap/pull/1402/files#r623753413